File: packages/event-bus-client/src/python/aeos_event_bus_client/topic.py
"""Tenant-scoped Kafka topic naming — mirrors TS topic.ts exactly."""
from __future__ import annotations


def topic_name(tenant_id: str, event_type: str) -> str:
    """
    Returns: aeos.{tenant_id}.{domain}.{event_type}
    Example: aeos.t-abc.telemetry.telemetry.span.received
    """
    domain = event_type.split(".")[0]
    return f"aeos.{tenant_id}.{domain}.{event_type}"


def topic_pattern(tenant_id: str, domain: str | None = None) -> str:
    """
    Returns a wildcard pattern for consumer subscription.
    Example: aeos.t-abc.telemetry.* or aeos.t-abc.*
    """
    if domain:
        return f"aeos.{tenant_id}.{domain}.*"
    return f"aeos.{tenant_id}.*"

File: packages/event-bus-client/src/topic.ts
import type { TenantId } from '@aeos/canonical-schema';
import type { AeosEvent } from '@aeos/canonical-schema';

export function topicName(tenantId: TenantId, eventType: AeosEvent['event_type']): string {
  const domain = eventType.split('.')[0];
  return `aeos.${tenantId}.${domain}.${eventType}`;
}

export function topicPattern(tenantId: TenantId, domain?: string): string {
  if (domain) {
    return `aeos.${tenantId}.${domain}.*`;
  }
  return `aeos.${tenantId}.*`;
}
